File: Problem002.hs
-- file: Project_Euler/Problem002.hs

fibs = 0 : 1 : zipWith (+) fibs (tail fibs)

showFibs n = take n fibs

sumEvenFibs :: Int -> Int
sumEvenFibs n = sum[x | x <- take n fibs, x `mod` 2 == 0]

main = putStrLn(show(sumEvenFibs 10))

File: Problem002.py
#Project Euler Problem #002

fibs = []
next_fib = 0
fibs.append(1)
fibs.append(2)
i = 2
while next_fib < 4000000:
    next_fib = fibs[i-1] + fibs[i-2]
    fibs.append(next_fib)
    i += 1

print(sum(x for x in fibs if x % 2 == 0))
